crates/bevy_state/src/state/mod.rs

```diff
@@ -619,4 +619,84 @@ mod tests {
             1
         );
     }
+
+    #[derive(Resource, Default, Debug)]
+    struct TransitionTracker(Vec<&'static str>);
+
+    #[test]
+    fn check_transition_orders() {
+        let mut world = World::new();
+        setup_state_transitions_in_world(&mut world, None);
+        EventRegistry::register_event::<StateTransitionEvent<SimpleState>>(&mut world);
+        EventRegistry::register_event::<StateTransitionEvent<SubState>>(&mut world);
+        EventRegistry::register_event::<StateTransitionEvent<TestComputedState>>(&mut world);
+        world.insert_resource(State(SimpleState::B(true)));
+        world.init_resource::<State<SubState>>();
+        world.insert_resource(State(TestComputedState::BisTrue));
+        let mut schedules = world.remove_resource::<Schedules>().unwrap();
+        let mut apply_changes = Schedule::new(StateTransition);
```

Comment:
This overrides the schedule that gets created by `setup_state_transitions_in_world`.
```suggestion
        let mut apply_changes = schedules.get_mut(StateTransition).unwrap();
```